### PR TITLE
Introduce toy runtime for testing inside substrate.

### PR DESCRIPTION
**Pull request by gavofyork**
Also:

- Refactor the runtime's support modules into a separate crate for deduping with the toy runtime in substrate.
- Removed some old code (`StaticHex` in favour of `hex_literal`).
- Rename `Slicable::to_vec` to `encode`.

### CODE REVIEW

File: polkadot/runtime/src/genesismap.rs [hunk 1]
```diff
@@ -52,32 +53,32 @@ impl GenesisConfig {
...
 		let wasm_runtime = include_bytes!("../wasm/genesis.wasm").to_vec();
 		vec![
-			(&b"gov:apr"[..], vec![].join(&self.approval_ratio)),
-			(&b"ses:len"[..], vec![].join(&self.session_length)),
-			(&b"ses:val:len"[..], vec![].join(&(self.validators.len() as u32))),
-			(&b"sta:wil:len"[..], vec![].join(&0u32)),
-			(&b"sta:spe"[..], vec![].join(&self.sessions_per_era)),
-			(&b"sta:vac"[..], vec![].join(&(self.validators.len() as u32))),
-			(&b"sta:era"[..], vec![].join(&0u64)),
+			(&b"gov:apr"[..], vec![].and(&self.approval_ratio)),
```
Comment (rphmeier) [inline polkadot/runtime/src/genesismap.rs]: wishlist: replace all these `vec![].and(x)` lines with something like `encode(x)`

File: substrate/test-runtime/src/transaction.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Transaction {
+	/// Who is sending.
+	pub from: AccountId,
+	/// Who to send to.
+	pub to: AccountId,
+	/// How much to send.
+	pub amount: u64,
+	/// How much to send.
+	pub nonce: u64,
```
Comment (rphmeier) [inline substrate/test-runtime/src/transaction.rs]: wrong comment on nonce

Comment (gavofyork) [reply substrate/test-runtime/src/transaction.rs]: fixed

**Comment by gavofyork** [comment]
cc @rphmeier

**Comment by rphmeier** [comment]
shouldn't it be built on #68 ?

**Pull request merged**